File: backend/src/models/reason.py
# models/reason.py
from typing import Optional
from datetime import datetime
from sqlmodel import SQLModel, Field, Relationship


class Reason(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    
    # Who is the reason from (the sender)
    from_user_id: int = Field(foreign_key="user.id")
    
    # Who is the reason to (the recipient)
    to_user_id: int = Field(foreign_key="user.id")
    
    # Content of the reason
    content: str = Field(max_length=500)
    
    # Timestamp
    created_at: datetime = Field(default_factory=datetime.utcnow)

    
    from_user: Optional["User"] = Relationship(
        sa_relationship_kwargs={"foreign_keys": "Reason.from_user_id"}
    )
    to_user: Optional["User"] = Relationship(
        sa_relationship_kwargs={"foreign_keys": "Reason.to_user_id"}
    )

File: mobile-app/yily_app/lib/models/reason.dart
class Reason {
  final int id;
  final int fromUserId;
  final int toUserId;
  final String content;
  final DateTime createdAt;

  Reason({
    required this.id,
    required this.fromUserId,
    required this.toUserId,
    required this.content,
    required this.createdAt,
  });

  factory Reason.fromJson(Map<String, dynamic> json) {
    return Reason(
      id: json['id'] as int,
      fromUserId: json['from_user_id'] as int,
      toUserId: json['to_user_id'] as int,
      content: json['content'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'from_user_id': fromUserId,
        'to_user_id': toUserId,
        'content': content,
        'created_at': createdAt.toIso8601String(),
      };
}